frontend-react/src/contexts/SessionContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { createContext, useContext } from "react";
+import { IOktaContext } from "@okta/okta-react/bundles/types/OktaContext";
+
+import useSessionStorage, {
+    SessionController,
+} from "../hooks/UseSessionStorage";
+import {
+    MembershipController,
+    useOktaMemberships,
+} from "../hooks/UseOktaMemberships";
+
+export interface ISessionContext {
+    memberships: MembershipController;
+    store: SessionController;
+}
+
+export type OktaHook = (_init?: Partial<IOktaContext>) => IOktaContext;
+
+interface ISessionProviderProps {
+    oktaHook: OktaHook;
+}
+
+export const SessionContext = createContext<ISessionContext>({
+    memberships: {} as MembershipController,
+    store: {} as SessionController,
+});
+
+// accepts `oktaHook` as a paremeter in order to allow mocking of this provider's okta based
```

Comment:
*paremeter = parameter

Reply:
I was actually going for `pearmeter` 🙃 will fix

Reply:
pear programming 🍐🍐